js/source/worker_tile.js

```diff
@@ -144,14 +141,9 @@ class WorkerTile {
             deps++;
             if (deps === 2) {
                 for (const bucket of this.symbolBuckets) {
-                    // Layers are shared and may have been used by a WorkerTile with a different zoom.
```

Comment:
This needs to happen immediately prior to `prepare` as well, since `prepare` uses layer properties. (Making this less fragile is tracked in @3479.)